core/include/vecmem/containers/impl/jagged_vector_buffer.ipp

```diff
@@ -15,170 +15,149 @@
+    memory_resource* host_access_resource)
+    : jagged_vector_buffer(::get_sizes(other), resource, host_access_resource) {
+
+}
+
+template <typename TYPE>
+jagged_vector_buffer<TYPE>::jagged_vector_buffer(
+    const std::vector<std::size_t>& sizes, memory_resource& resource,
+    memory_resource* host_access_resource)
+    : base_type(sizes.size(), nullptr),
+      m_outer_memory(::allocate_jagged_buffer_outer_memory<TYPE>(
+          (host_access_resource == nullptr ? 0 : sizes.size()), resource)),
+      m_outer_host_memory(::allocate_jagged_buffer_outer_memory<TYPE>(
+          sizes.size(),
+          (host_access_resource == nullptr ? resource
+                                           : *host_access_resource))),
+      m_inner_memory(
+          ::allocate_jagged_buffer_inner_memory<TYPE>(sizes, resource, false)) {
+  // Point the base class at the newly allocated memory.
+  base_type::m_ptr =
+      ((host_access_resource != nullptr) ? m_outer_memory.get()
+                                         : m_outer_host_memory.get());
+
+  // Set up the host accessible memory array.
+  std::ptrdiff_t ptrdiff = 0;
+  for (std::size_t i = 0; i < sizes.size(); ++i) {
+    new (host_ptr() + i) value_type(
+        sizes[i], reinterpret_cast<TYPE*>(m_inner_memory.get() + ptrdiff));
+    ptrdiff += sizes[i] * sizeof(TYPE);
+  }
+}
+
+template <typename TYPE>
+jagged_vector_buffer<TYPE>::jagged_vector_buffer(
+    const std::vector<std::size_t>& sizes,
+    const std::vector<std::size_t>& capacities, memory_resource& resource,
+    memory_resource* host_access_resource)
+    : base_type(sizes.size(), nullptr),
+      m_outer_memory(::allocate_jagged_buffer_outer_memory<TYPE>(
+          (host_access_resource == nullptr ? 0 : sizes.size()), resource)),
+      m_outer_host_memory(::allocate_jagged_buffer_outer_memory<TYPE>(
+          sizes.size(),
+          (host_access_resource == nullptr ? resource
+                                           : *host_access_resource))),
+      m_inner_memory(::allocate_jagged_buffer_inner_memory<TYPE>(
+          capacities, resource, true)) {
+  // Some sanity check.
+  assert(sizes.size() == capacities.size());
+
+  // Point the base class at the newly allocated memory.
+  base_type::m_ptr =
+      ((host_access_resource != nullptr) ? m_outer_memory.get()
+                                         : m_outer_host_memory.get());
+
+  // Set up the host accessible memory array.
+  std::ptrdiff_t ptrdiff = 0;
+  for (std::size_t i = 0; i < capacities.size(); ++i) {
+    new (host_ptr() + i) value_type(
+        capacities[i],
+        reinterpret_cast<typename value_type::size_pointer>(
+            m_inner_memory.get() + ptrdiff),
+        reinterpret_cast<TYPE*>(m_inner_memory.get() + ptrdiff +
+                                sizeof(typename value_type::size_type)));
+    ptrdiff +=
+        capacities[i] * sizeof(TYPE) + sizeof(typename value_type::size_type);
+  }
```

Comment:
This is an example of some code that becomes significantly less readable with these settings, in my opinion.

Reply:
It does, I don't disagree.

Reply:
One can increase the line length to alleviate this to some extent.